src/main/java/gameframework/particles/Particle.java

```diff
@@ -0,0 +1,167 @@
+package gameframework.particles;
+
+import java.awt.Color;
+import java.awt.Rectangle;
+
+import gameframework.particles.behaviors.ParticleBehavior;
+
+/**
+ * Stores every useful information about a single particle
+ * like its position on the screen, its color...
+ */
+public class Particle {
+	protected double x;
+	protected double y;
+	protected double velocityY;
+	protected int timeAlive;
+	protected Color color;
+	protected double randomness;
+	protected int id;
+	protected int width;
+	protected int height;
+	protected ParticleBehavior behavior;
+	
+	/**
+	 * Constructor
+	 * @param id Id of the particle
+	 * @param color Color of the particle
+	 * @param x Initial x position of the particle on the screen
+	 * @param y Initial y position of the particle on the screen
+	 * @param width Initial width of the particle on the screen
+	 * @param height Initial height of the particle on the screen
+	 * @param behavior Behavior used to update the particle
+	 */
+	public Particle(int id, Color color, int x, int y, int width, int height, ParticleBehavior behavior) {
+		this.x = x;
+		this.y = y;
+		this.width = width;
+		this.height = height;
+		this.behavior = behavior;
+		this.randomness = Math.random();
+		this.color = color;
+		this.id = id;
+	}
+	
+	/**
+	 * Constructor
+	 * @param id Id of the particle
```

Comment:
which Id should a user pass? Why can't the class take care of generating IDs itself?